addons/account/models/reconciliation_widget.py

```diff
@@ -417,14 +445,22 @@ def _domain_move_lines(self, search_str):
         return str_domain
 
     @api.model
-    def _domain_move_lines_for_reconciliation(self, st_line, aml_accounts, partner_id, excluded_ids=None, search_str=False):
+    def _domain_move_lines_for_reconciliation(self, st_line, aml_accounts, partner_id, excluded_ids=[], search_str=False):
         """ Return the domain for account.move.line records which can be used for bank statement reconciliation.
 
             :param aml_accounts:
             :param partner_id:
             :param excluded_ids:
             :param search_str:
         """
+        to_check = self._context.get('mode') == 'to_check'
+        
+        if to_check:
+            to_check_excluded = self.env['account.move.line'].search([
+                ('move_id.to_check', '=', True),
+                ('full_reconcile_id', '=', False),
```

Comment:
it's the third time you need to search account.move.line that are 'toc_check', this lightens a signal that it needs a function returning the domain to use for the search. Doesn't it?